examples/src/main/java/com/mapbox/navigation/examples/core/RouteAlertsActivity.kt

```diff
@@ -0,0 +1,187 @@
+package com.mapbox.navigation.examples.core
+
+import android.annotation.SuppressLint
+import android.graphics.Color
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.core.constants.Constants
+import com.mapbox.geojson.Feature
+import com.mapbox.geojson.FeatureCollection
+import com.mapbox.geojson.LineString
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory
+import com.mapbox.mapboxsdk.location.modes.RenderMode
+import com.mapbox.mapboxsdk.style.layers.LineLayer
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.base.trip.model.alert.RouteAlert
+import com.mapbox.navigation.base.trip.model.alert.RouteAlertType
+import com.mapbox.navigation.base.trip.model.alert.TunnelEntranceAlert
+import com.mapbox.navigation.base.trip.model.alert.toLineString
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.replay.MapboxReplayer
+import com.mapbox.navigation.core.replay.ReplayLocationEngine
+import com.mapbox.navigation.core.replay.route.ReplayRouteMapper
+import com.mapbox.navigation.core.trip.session.RouteAlertsObserver
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+import com.mapbox.navigation.examples.R
+import com.mapbox.navigation.examples.utils.Utils
+import com.mapbox.navigation.ui.camera.NavigationCamera
+import com.mapbox.navigation.ui.map.NavigationMapboxMap
+import kotlinx.android.synthetic.main.activity_replay_route_layout.mapView
+import kotlinx.android.synthetic.main.activity_route_alerts.*
+import kotlin.math.absoluteValue
+import kotlin.math.roundToInt
+
+class RouteAlertsActivity : AppCompatActivity() {
+    private val directionsRoute: DirectionsRoute by lazy {
+        val directionsResponseJson = resources.openRawResource(
+            R.raw.route_response_tunnel_driving
+            // R.raw.mock_response_all_passive_maneuvers_polyline6
```

Comment:
nit: mind adding a comment or put it in a function, some indicator that it's here as an example option